internal/error.go

```diff
@@ -842,79 +869,95 @@ func ConvertErrorToFailure(err error, dc converter.DataConverter) *failurepb.Fai
 	} else {
 		failure.Message = err.Error()
 	}
-
-	switch err := err.(type) {
-	case *ApplicationError:
-		failureInfo := &failurepb.ApplicationFailureInfo{
-			Type:         err.errType,
-			NonRetryable: err.nonRetryable,
-			Details:      convertErrDetailsToPayloads(err.details, dc),
-		}
-		failure.FailureInfo = &failurepb.Failure_ApplicationFailureInfo{ApplicationFailureInfo: failureInfo}
-	case *CanceledError:
-		failureInfo := &failurepb.CanceledFailureInfo{
-			Details: convertErrDetailsToPayloads(err.details, dc),
-		}
-		failure.FailureInfo = &failurepb.Failure_CanceledFailureInfo{CanceledFailureInfo: failureInfo}
-	case *PanicError:
-		failureInfo := &failurepb.ApplicationFailureInfo{
-			Type: getErrType(err),
-		}
-		failure.FailureInfo = &failurepb.Failure_ApplicationFailureInfo{ApplicationFailureInfo: failureInfo}
-		failure.StackTrace = err.StackTrace()
-	case *workflowPanicError:
-		failureInfo := &failurepb.ApplicationFailureInfo{
-			Type:         getErrType(&PanicError{}),
-			NonRetryable: true,
+	originalError := err
+
+Loop:
+	for ; err != nil; err = errors.Unwrap(err) {
```

Comment:
This is gonna swallow errors. What would have happened before if I had `*MyError` that wrapped `*MyOtherError` that wrapped `*ApplicationError` that had details? Before it'd be a `MyError` w/ cause `MyOtherError` with cause `ApplicationFailure` of whatever type and details. Now, this would be moving those details to the top-level error improperly.

Why do we have to change this function at all? If you are concerned about unwrapping your update error specifically, unwrap it specifically where you handle it.

Reply:
The idea was for those cases where one of these well-known error types gets wrapped up in a bypassDurabilityError, that error wouldn't become the "top" Failure. But as you see with the integration test failures, there are other places that are depend on the structure returned here and how it maps to the concrete error types so I'll need to revert this bit.

Reply:
Yeah, I think we can leave this function alone and you can just unwrap your specific bypass-durability error when handling update validation result

Reply:
Resolution: will remain unchanged.